tests/test_project.cairo

```diff
@@ -175,3 +197,104 @@ fn test_project_set_vintage_status() {
     let vinatge: CarbonVintage = carbon_credits.get_specific_carbon_vintage(2025);
     assert(vinatge.cc_status == CarbonVintageType::Audited, 'Error of status');
 }
+
+/// Test balance_of
+#[test]
+fn test_project_balance_of() {
+    let owner_address: ContractAddress = contract_address_const::<'owner'>();
+    let (project_address, _) = default_setup();
+    let absorber = IAbsorberDispatcher { contract_address: project_address };
+    let carbon_credits = ICarbonCreditsHandlerDispatcher { contract_address: project_address };
+    let project_contract = IProjectDispatcher { contract_address: project_address };
+
+    start_prank(CheatTarget::One(project_address), owner_address);
+
+    assert(absorber.is_setup(), 'Error during setup');
+
+    let share = 33 * MULT_ACCURATE_SHARE / 100;
+    mint_utils(project_address, owner_address, share);
+
+    let supply_vintage_2025 = carbon_credits.get_specific_carbon_vintage(2025).cc_supply;
+    let expected_balance = 3300000;
+    let balance = project_contract.balance_of(owner_address, 2025);
+    assert(balance == expected_balance.into(), 'Error of balance');
+}
+
+#[test]
+fn test_transfer_without_loss() {
+    let owner_address: ContractAddress = contract_address_const::<'owner'>();
+    let (project_address, _) = default_setup();
+    let absorber = IAbsorberDispatcher { contract_address: project_address };
+    let project_contract = IProjectDispatcher { contract_address: project_address };
+
+    start_prank(CheatTarget::One(project_address), owner_address);
+
+    assert(absorber.is_setup(), 'Error during setup');
+
+    let share = 33 * MULT_ACCURATE_SHARE / 100;
+    mint_utils(project_address, owner_address, share);
+
+    let expected_balance = 3300000;
+    let balance = project_contract.balance_of(owner_address, 2025);
+    assert(balance == expected_balance.into(), 'Error of balance');
+
+    let receiver_address: ContractAddress = contract_address_const::<'receiver'>();
+    let receiver_balance = project_contract.balance_of(receiver_address, 2025);
+    assert(receiver_balance == 0, 'Error of balance');
+
+    project_contract
+        .safe_transfer_from(owner_address, receiver_address, 2025, 3300000.into(), array![].span());
```

Comment:
is there a way to avoid safe_transfer?

Reply:
The safe methods here are not safe methods, we did fork the erc1155 code and removed the safe logic because OZ standard removed the "normal" (not safe) transfer method from their interface.